src/agent/api/src/apisvc.c

```diff
@@ -361,12 +391,25 @@ static void* aduc_apisvc_thread_proc(void* arg)
             }
         }
 
+        if (sent > 0)
+        {
+            fsync(writefifo); // attempt to flush to the pipe before closing
+            usleep(50000); // 50 msec delay to allow client to read before closing the fifo
+        }
+
```

Comment:
Can this cause a potential race condition? (e.g., where the reader make take longer than 50000 msec to read? (or currently doing something else and not currently reading the data?)